app/models/card.rb

```diff
@@ -3,29 +3,70 @@ class Card < ActiveRecord::Base
   belongs_to :user
   accepts_nested_attributes_for :deck
 
-  before_validation :set_default_review_date
+  before_create :set_defaults
 
   has_attached_file :picture, styles: { medium: "360x360>", thumb: "50x50>" }, default_url: "/images/placeholder.png"
   validates_attachment_content_type :picture, content_type: /\Aimage\/.*\Z/
-  validates :original_text, :translated_text, :review_date, presence: true
+  validates :original_text, :translated_text, presence: true
   validates :deck_id, presence: true
   validate :not_equal_fields
-  scope :for_review, -> { where("review_date < ?", Date.today) }
-
-  def move_review_date
-    self.review_date = Date.today + 3.days
-    save
-  end
+  scope :for_review, -> { where("review_date < ?", Time.now) }
 
   def check_translation(translation)
     if translated_text == clear(translation)
-      move_review_date
+      review_true
       return true
     else
+      review_false
       return false
     end
   end
 
+  def review_true
+    current_level = self.review_level
+    self.error_counter = 0
+    if current_level < 5
+      self.review_level += 1
+      new_date = Card.leitner_step(review_level)
+      self.review_date = new_date
+    end
+    save
+  end
+
+  def review_false
+    current_level = self.review_level
+    errors = self.error_counter
+    card_changed = false
+    if current_level > 0
+      if errors == 3
+        self.review_level -= 1
+        self.error_counter = 0
+        save
+      else
+        self.error_counter += 1
+        save
+      end
+    end
+  end
+
+  def self.leitner_step(review_step)
+    today = Time.now
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.